groups/csa/csamisc/csamisc_emptystringinitialisation.cpp

```diff
@@ -0,0 +1,140 @@
+#include <clang/ASTMatchers/ASTMatchers.h>
+#include <clang/ASTMatchers/ASTMatchFinder.h>
+#include <clang/Lex/Lexer.h>
+
+#include <clang/AST/Stmt.h>
+#include <csabase_analyser.h>
+#include <csabase_debug.h>
+#include <csabase_diagnostic_builder.h>
+#include <csabase_registercheck.h>
+#include <csabase_report.h>
+#include <csabase_util.h>
+#include <csabase_visitor.h>
+#include <set>
+
+#include <iostream>
+
+using namespace csabase;
+using namespace clang;
+using namespace clang::ast_matchers;
+
+static std::string const check_name("emptystringinit");
+
+namespace {
+	
+struct data {
+
+};
+
+
+struct report : Report<data>
+{
+    INHERIT_REPORT_CTOR(report, Report, data);
+
+    void stringInit(BoundNodes const & nodes);
+    void operator()();
+};
+
+
+//matches string initialization as a variable or in an initialization list
+internal::DynTypedMatcher dyn_matchEmptyStringInit()
+{
+	
+    internal::DynTypedMatcher matcher = findAll( decl( eachOf( 
+													varDecl( hasInitializer( 
+												    	hasDescendant ( stringLiteral().bind("string") ) 
+											    		)).bind("varDecl"),
+													constructorDecl( forEachConstructorInitializer( ctorInitializer(withInitializer(
+													hasDescendant( stringLiteral().bind("string")))).bind("varInit"))) ))); 
+    return matcher; 
+}
+
+
+void report::operator()()
+{
+    MatchFinder finder;
+    csabase::OnMatch<report, &report::stringInit> m(this);
+    finder.addDynamicMatcher(dyn_matchEmptyStringInit(), &m);
+    finder.match(*d_analyser.context()->getTranslationUnitDecl(), *d_analyser.context());
+}
+
+
+void report::stringInit(BoundNodes const & nodes)
```

Comment:
There is quite a bit of code duplication, could you please refactor so that in each if branch you obtain/calculate location for report/replacement, range to replace and replacement string.

In addition, checking if the string literal is empty can be done first thing in the function if not empty then just return.